model/fieldtypes/ForeignKey.php

```diff
@@ -37,17 +37,16 @@ public function scaffoldFormField($title = null, $params = null) {
 			$field = new UploadField($relationName, $title);
 		} else {
 			$titleField = (singleton($hasOneClass)->hasField('Title')) ? "Title" : "Name";
-			// Use a SQL query instead of the ORM to save on PHP memory, DB objects and queries
-			$list = array();
-			$sqlQuery = new SQLSelect();
-			$sqlQuery->setFrom($hasOneClass);
-
-			$result = $sqlQuery->execute();
-			foreach ($result as $row) {
-				$list[$row['ID']] = $row[$titleField];
+			$list = DataList::create($hasOneClass);
+			// Don't scaffold a dropdown for large tables, as making the list concrete
+			// might exceed the available PHP memory in creating too many DataObject instances
+			if($list->count() < 100) {
+				$field = new DropdownField($this->name, $title, $list->map('ID', $titleField));
+				$field->setEmptyString(' ');
+			} else {
+				$field = null;
```

Comment:
So in the CMS if a object has over 100 records we just won't show a field? I don't think that's acceptable. With the new work using generators instead of iterators building large lists for forms should be much more efficient now...

Just removing a field at (what will appear to be) random isn't a good experience for a CMS user...